### PR TITLE
Initial commit of mathsteps textarea demo.

### PR DESCRIPTION
**Pull request by tkosan**
Instructions for how to use [browserify](https://github.com/substack/node-browserify) to build a version of mathsteps that runs in the browser is in `lib/mathstepsWeb.js`.

### CODE REVIEW

**Comment by evykassirer** [comment]
sweeet!

eventually I'd like this to work with [github pages](https://pages.github.com/) so we'd need `mathsteps.js` built in the repo

I'm a bit busy for the next few days, but I'll take a look soon!

File: lib/mathstepsWeb.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    try {
+      const steps = mathsteps.solveEquation(equationString, false);
+      if (steps.length != 0) {
+        result.steps = steps;
+      }
+      else {
+        result.error = "Unable to solve equation";
+      }
+    }
+    catch (e) {
```
Comment (evykassirer) [inline lib/mathstepsWeb.js]: good call on the try catch, sometimes errors happen (ideally never though)

File: lib/demos/textArea/mathsteps_text_area.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      <P><INPUT TYPE="button" VALUE="Clear Input Text Box" name="clearInputButton" onClick="this.form.inputTextField.value=''"></p>
+      <P><INPUT TYPE="button" VALUE="Clear Result Text Area" name="clearResultButton" onClick="this.form.resultTextArea.value=''"></p>
+    </FORM>
+    
+    <script type="text/javascript" language="javascript" src="mathsteps.js"></script>
+
+  </body>
+
+</html>
+
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: don't need this extra trailing line

File: lib/demos/textArea/mathsteps_text_area.html [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        if (key == 13) solve(document.getElementById('inputTextField').value);
+      };
+      
+    </script>
+
+  </head>
+
+  <body>
+
+    <h2>Mathsteps Demo v.01</h2>
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: this says v.01 but the page title says v.02

which do you wanna use?

File: lib/demos/textArea/mathsteps_text_area.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
+<!DOCTYPE html>
+
+<html>
+  <head>
+  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
+  <title>Mathsteps demo v.02</title> 
+
+    <script type="text/javascript" language="javascript">
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: can you make this a separate JS file that's only linked to the html through the script tag?

File: lib/demos/textArea/mathsteps_text_area.html [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    
+      function solve(expression)
+      {
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: open brace on same line pls :)

File: lib/demos/textArea/mathsteps_text_area.html [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        appendText("\r\r");
+        document.getElementById("resultTextArea").scrollTop = document.getElementById("resultTextArea").scrollHeight; 
+      }
+    
+      function appendText(text) {
+        var obj=document.getElementById("resultTextArea");
+        obj.value += text;
+      }
+    
+      var handleKeyPress = function(evt){
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: what does this function do? I'm not familiar with what you're doing in the next two lines

adding a docstring would be helpful

File: lib/demos/textArea/mathsteps_text_area.html [hunk 7]
```diff
@@ -0,0 +1,70 @@
+<!DOCTYPE html>
+
+<html>
+  <head>
+  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
+  <title>Mathsteps demo v.02</title> 
+
+    <script type="text/javascript" language="javascript">
+    
+      function solve(expression)
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: can you add a docstring?

File: lib/demos/textArea/mathsteps_text_area.html [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      };
+      
+    </script>
+
+  </head>
+
+  <body>
+
+    <h2>Mathsteps Demo v.01</h2>
+    mathsteps project website <a href="https://github.com/socraticorg/mathsteps">https://github.com/socraticorg/mathsteps</a>
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: can you change this text to be "Source code for mathsteps: "

File: lib/demos/textArea/mathsteps_text_area.html [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    
+      function appendText(text) {
+        var obj=document.getElementById("resultTextArea");
+        obj.value += text;
+      }
+    
+      var handleKeyPress = function(evt){
+        var key = evt.which || evt.charCode || evt.keyCode || 0;
+        if (key == 13) solve(document.getElementById('inputTextField').value);
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: my security friend just told me we should sanitize the input - idk if anything bad could happen, but it doesn't hurt :)

File: lib/demos/textArea/mathsteps_text_area.html [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+  </head>
+
+  <body>
+
+    <h2>Mathsteps Demo v.01</h2>
+    mathsteps project website <a href="https://github.com/socraticorg/mathsteps">https://github.com/socraticorg/mathsteps</a>
+    
+    <noscript>
+      <div style="width: 22em; position: absolute; left: 50%; margin-left: -11em; color: red; background-color:white; border: 1px solid red; padding: 4px; font-family: sans-serif">
```
Comment (evykassirer) [inline lib/demos/textArea/mathsteps_text_area.html]: can you make a stylesheet?

**Comment by tkosan** [comment]
@evykassirer my main purpose for submitting this code was as a proof of concept. At this point it would probably be best if someone who is stronger in front end development than I am to use this code as a starting point for the actual first demo :smile:

**Comment by evykassirer** [comment]
hehe fair enough ^_^ let's keep this PR open and someone (maybe me, we'll see!) can clone your code and move some stuff around and push that to start

thanks for the starter stuff :D

File: lib/demos/textArea/mathsteps_text_area.html (hunk 6 above)
Comment (tkosan) [inline lib/demos/textArea/mathsteps_text_area.html]: This code provides a browser-independent way for detecting when the <Enter> key is pressed.

**Pull request closed without merging**